### PR TITLE
Add API dropdowns

### PR DESCRIPTION
**Pull request by colleenmcginnis**
This WIP PR does a few things related to API reference links throughout this repo: 

- [x] Fixes https://github.com/mapbox/android-docs/issues/814
- [x] Fixes https://github.com/mapbox/android-docs/issues/868
- [x] Adds an update script (maybe)
- [x] Adds redirects from `/android/api/{product}` to `/android/api/{product}/{current_version}` so `/android/api/{product}` is an ever green link 🌲 

The needs for each product in this repo is a little different so the current implementation is a bit complex. @katydecorah I'd love your thoughts on the current approach and would welcome any suggestions for simplifying/organizing.

 | Product | Proposed dropdown |
|---|---|
| **Maps** | screen shot 2019-03-01 at 12 56 44 pm |
| **Navigation** | screen shot 2019-03-01 at 12 57 54 pm |
| **Java** |  screen shot 2019-03-01 at 1 00 09 pm |
| **Plugins** | screen shot 2019-03-01 at 1 00 34 pm |
| **Core** | screen shot 2019-03-01 at 1 00 50 pm |
| **Vision + Vision AR**  | TBD |

cc @langsmith @malwoodsantoro @riastrad

### CODE REVIEW

**Comment by katydecorah** [comment]
@colleenmcginnis I updated dr-ui so we can use the versionSort function and also organized the dropdown item options into a config file to keep like items together.

Right now the dropdown menu is not staying open long enough to click on an item. Currently debugging 🐛

**Comment by katydecorah** [comment]
@colleenmcginnis I decided to back track and use `@mapbox/react-popover-trigger` instead of mr-ui's PopoverTrigger, here's why:

1. `@mapbox/react-popover-trigger` works beautiful on hover to open the menu and mr-ui's version does not (at least not right now).
2. I had mentioned about making the menus click-to-open, but I like the hover-to-open the menu, instead of click. It feels so much smoother!
3. Using `@mapbox/react-popover-trigger` unblocks this PR! Down the road we can either decide to  make it a dr-ui component or figure out how to update the mr-ui version.

Let me know what you think!

**Comment by colleenmcginnis** [comment]
>I decided to back track and use @mapbox/react-popover-trigger instead of mr-ui's PopoverTrigger

I think this is ok as a short-term solution. Thanks for digging in! 

>Down the road we can either decide to make it a dr-ui component or figure out how to update the mr-ui version.

I think you should definitely work on updating the mr-ui version so other teams can use it too. 🙂

File: src/components/top-nav-tabs.js [hunk 1]
```diff
@@ -32,66 +32,103 @@ class TopNavTabs extends React.Component {
...
-                {link.label}{' '}
-                {index === 0 ? <span> &mdash; latest stable</span> : ''}{' '}
-              </span>
-              <a
-                className="color-gray color-gray-dark-on-hover block ml12"
-                href={`/android/api/navigation-sdk/navigation/${
-                  link.label
-                }/index.html`}
+
+    const config = {
```
Comment (colleenmcginnis) [inline src/components/top-nav-tabs.js]: Wow @katydecorah you are brilliant. 💡

**Comment by colleenmcginnis** [comment]
Thank you for pairing on the update script @katydecorah!! 🍐 This is ready for another look. 👀

File: scripts/update.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+};
+
+/* Validate the arguments */
+const validateProduct = Object.keys(config).indexOf(PRODUCT) < 0;
+const validateVersion = !/[0-9]+\.[0-9]+\.[0-9]+-?([beta]+)?([rc]+)?\.?[0-9]?/.test(
+  VERSION
+);
+if (validateProduct || validateVersion) {
+  if (validateProduct)
+    console.log(
```
Comment (katydecorah) [inline scripts/update.js]: Another option is to use `throw new Error()` and then remove `return` on line 43. `throw` statements will stop the code from executing and will display a little more detail in the error message about where in the code you're receiving an error.

File: scripts/update.js [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+      `❌ Invalid product type: ${PRODUCT}. Use either \`maps\` or \`navigation\`.`
+    );
+  if (validateVersion)
+    console.log(
+      `❌ Invalid version number: ${VERSION}. Use \`X.X.X\`, where X is an integer. Optional: append the version number with \`-beta.X\` or \`-rc.X\`.`
+    );
+  return;
+}
+
+/* Read file contens and write to those files */
```
Comment (katydecorah) [inline scripts/update.js]:
```suggestion
/* Read file contents and write to those files */
```

File: scripts/update.js (hunk 2 above)
Comment (colleenmcginnis) [reply scripts/update.js]: 🤓 I learn something new every day I work with @katydecorah.

**Pull request merged**